packages/bpk-component-split-input/src/BpkSplitInput.tsx

```diff
@@ -202,11 +216,12 @@ class BpkSplitInput extends Component {
           large={large}
           value={inputValue && inputValue[index]}
           placeholder={placeholder}
-          onChange={this.handleOnChange}
+          {...rest}
```

Comment:
This was a conscious decision to put rest there so we override the below components based on how we handle them. rest at the end was making unwanted changes.

Reply:
Ok, maybe we need to check this out more, as rest should always come last. Could you clarify the unwanted change this would be making?

Could it be something we are passing in from the examples that overrides this

Reply:
As far as I understand we want to handle props as we have described in each function for:
```
          onInput={this.handleOnChange}
          onKeyDown={this.handleOnKeyDown}
          onPaste={this.handleOnPaste}
          onFocus={(e: FocusEvent<HTMLInputElement>) => this.handleOnFocus(e, index)}
          onChange={this.handleOnChange}
```
But if we use `rest` at the end, it overwrites our functions and how we want to handle our props.

Reply:
So this Component has all these custom handle event functions for managing behaviour. 
If we allow parent onChange functions to override this components onEvent behaviour it could break.. Though I suppose it's up to the consumer... I think I may be wrong about this and rest can be at the end.

Reply:
Alright ... If I move {...rest} down, it means they are overwiring the handle fuctions which makes the handle functions irrelevant and creating error below: 
![image](https://github.com/user-attachments/assets/968546e0-e730-40c1-94c7-a8999290ad08)
How do we handle that?

Reply:
That's only if the consumer specifies the on{Event} functions.  Which I suppose the onChange is a prop. You can define a default value for onChange? or is it that onChange is set in handleOnChange?

Reply:
If we defin onChange above then this error should go away? and we can specify if `onChange ? onChange : this.handleOnChange`

Reply:
Well turns out this prop was poorly named as it's purpose wasn't to be passed down to the input field it's to just give the consumer a way to add their function to the existing logic.